indico/modules/events/editing/client/js/editing/index.jsx

```diff
@@ -33,52 +29,31 @@ document.addEventListener('DOMContentLoaded', async () => {
     document.querySelector('div.main-breadcrumb').getBoundingClientRect().height;
   document.body.style.setProperty('--header-height', headerHeight);
 
-  const eventId = parseInt(editingElement.dataset.eventId, 10);
   const eventTitle = editingElement.dataset.eventTitle;
-  const contributionId = parseInt(editingElement.dataset.contributionId, 10);
-  const contributionCode = editingElement.dataset.contributionCode;
-  const editableType = editingElement.dataset.editableType;
-
-  let fileTypes, tags;
-
-  try {
-    const [fileTypeResponse, tagResponse] = await Promise.all([
-      indicoAxios.get(fileTypesURL({confId: eventId, type: editableType})),
-      indicoAxios.get(tagsURL({confId: eventId})),
-    ]);
-    fileTypes = camelizeKeys(fileTypeResponse.data);
-    tags = camelizeKeys(tagResponse.data);
-  } catch (e) {
-    handleAxiosError(e);
-    return;
-  }
-
-  const store = createReduxStore(
-    'editing-timeline',
-    {timeline: reducer},
-    {
-      staticData: {
-        eventId,
-        contributionId,
-        contributionCode,
-        editableType,
-        fileTypes,
-        tags,
-        editableDetailsURL: editableDetailsURL({
-          confId: eventId,
-          contrib_id: contributionId,
-          type: editableType,
-        }),
-      },
-    }
-  );
 
   ReactDOM.render(
-    <Provider store={store}>
-      <EditingView eventId={eventId} eventTitle={eventTitle}>
-        <Timeline />
-      </EditingView>
-    </Provider>,
+    <Router>
+      <Switch>
+        <Route
+          exact
+          path={routerPathFromFlask(timelineURL, ['confId', 'contrib_id', 'type'])}
+          component={() => (
```

Comment:
Even though this works, `component` is meant to be used with just that, e.g. `component={ReduxTimeline}`. But since you don't rely on any render props, you can just pass the JSX you return here as children to the `Route`.